pjsip/src/pjsua-lib/pjsua_pres.c

```diff
@@ -1964,6 +1969,11 @@ static void unsubscribe_buddy_presence(pjsua_buddy_id buddy_id)
     PJ_LOG(5,(THIS_FILE, "Buddy %d: unsubscribing..", buddy_id));
     pj_log_push_indent();
 
+    /* Set route-set */
+    if (!pj_list_empty(&acc->route_set)) {
+	pjsip_dlg_set_route_set(buddy->dlg, &acc->route_set);
+    }
```

Comment:
Shouldn't this already be set when subscribing the presence?

Reply:
If the `buddy->dlg` is the same and if route set is already set, yes it should be.
I just checked again, and the `Route` header is not set in the un`SUBSCRIBE` request.

Maybe this is not the proper way to fix it : I can investigate when I have more time if the dialog is the same, and if it is, why the route set is unset.